front/lib/resources/mcp_server_view_resource.ts

```diff
@@ -0,0 +1,310 @@
+    ids: string[]
+  ): Promise<Result<MCPServerViewResource[], DustError>> {
+    const viewModelIds = removeNulls(ids.map((id) => getResourceIdFromSId(id)));
+    if (viewModelIds.length !== ids.length) {
+      return new Err(new DustError("invalid_id", "Invalid id"));
+    }
+
+    const views = await this.baseFetch(auth, {
+      where: {
+        id: {
+          [Op.in]: viewModelIds,
+        },
+      },
+    });
+
+    if (views.length !== ids.length) {
+      return new Err(
+        new DustError(
+          "resource_not_found",
+          ids.length === 1 ? "View not found" : "Some views were not found"
+        )
+      );
+    }
+
+    return new Ok(views);
+  }
+
+  static async fetchByModelPk(auth: Authenticator, id: ModelId) {
+    const views = await this.fetchByModelIds(auth, [id]);
+
+    if (views.length !== 1) {
+      return null;
+    }
+
+    return views[0];
+  }
+
+  static async fetchByModelIds(auth: Authenticator, ids: ModelId[]) {
+    const views = await this.baseFetch(
+      auth,
+
+      {
+        where: {
+          id: {
+            [Op.in]: ids,
+          },
+        },
+      }
+    );
+
+    return views ?? [];
+  }
+
+  static async listByWorkspace({
+    auth,
+  }: {
+    auth: Authenticator;
+  }): Promise<MCPServerViewResource[]> {
+    return this.baseFetch(auth);
+  }
+
+  // Deletion.
+
+  async delete(
+    auth: Authenticator,
+    { transaction }: { transaction?: Transaction } = {}
+  ): Promise<Result<undefined, Error>> {
+    assert(auth.isAdmin(), "Only the admin can delete an MCP server view");
+
+    try {
+      await this.model.destroy({
+        where: {
+          id: this.id,
+        },
+        transaction,
+      });
+      return new Ok(undefined);
+    } catch (err) {
+      return new Err(err as Error);
+    }
+  }
+
+  // Deletion.
+
+  protected async softDelete(
+    auth: Authenticator,
+    transaction?: Transaction
+  ): Promise<Result<number, Error>> {
+    const deletedCount = await MCPServerView.destroy({
+      where: {
+        workspaceId: auth.getNonNullableWorkspace().id,
+        id: this.id,
+      },
+      transaction,
+      hardDelete: false,
+    });
+
+    return new Ok(deletedCount);
+  }
+
+  async hardDelete(
+    auth: Authenticator,
+    transaction?: Transaction
+  ): Promise<Result<number, Error>> {
+    // Delete agent configurations elements pointing to this data source view.
+    await AgentMCPServerConfiguration.destroy({
+      where: {
+        mcpServerViewId: this.id,
+      },
+      transaction,
+    });
+
+    const deletedCount = await MCPServerView.destroy({
+      where: {
+        workspaceId: auth.getNonNullableWorkspace().id,
+        id: this.id,
+      },
+      transaction,
+      // Use 'hardDelete: true' to ensure the record is permanently deleted from the database,
+      // bypassing the soft deletion in place.
+      hardDelete: true,
+    });
+
+    return new Ok(deletedCount);
+  }
+
+  async getRemoteMCPServer(
```

Comment:
I would automatically load that if relevant as part of the initial fetch? We have prior art for that.